src/clj/puppetlabs/services/jruby/jruby_puppet_service.clj

```diff
@@ -43,28 +38,25 @@
             pool-context (create-pool jruby-config)]
         (-> context
             (assoc :pool-context pool-context)
-            (assoc :borrow-timeout (:borrow-timeout config))
-            (assoc :event-callbacks (atom []))
             (assoc :environment-class-info-tags (atom {}))))))
+
   (stop
    [this context]
-   (let [{:keys [pool-context]} (tk-services/service-context this)
-         on-complete (promise)]
-     (log/debug "Beginning flush of JRuby pools for shutdown")
-     (jruby-agents/send-flush-pool-for-shutdown! pool-context on-complete)
-     @on-complete
-     (log/debug "Finished flush of JRuby pools for shutdown"))
+   (let [{:keys [pool-context]} (tk-services/service-context this)]
+     (jruby-core/flush-pool-for-shutdown! pool-context))
    context)
 
   (borrow-instance
     [this reason]
-    (let [{:keys [pool-context borrow-timeout event-callbacks]} (tk-services/service-context this)]
-      (jruby-core/borrow-from-pool-with-timeout pool-context reason @event-callbacks)))
+    (let [{:keys [pool-context]} (tk-services/service-context this)
```

Comment:
Sure! I was really lazy and just mostly kept things as they were as much as possible, but feel free to amend this as you wish.